ext/standard/php_uuid.h

```diff
@@ -0,0 +1,163 @@
+	PHP_UUID_VARIANT_FUTURE_RESERVED = 3,
+} php_uuid_variant;
+
+static const uint8_t PHP_UUID_VERSION_1_TIME_BASED      = 1;
+static const uint8_t PHP_UUID_VERSION_2_DCE_SECURITY    = 2;
+static const uint8_t PHP_UUID_VERSION_3_NAME_BASED_MD5  = 3;
+static const uint8_t PHP_UUID_VERSION_4_RANDOM          = 4;
+static const uint8_t PHP_UUID_VERSION_5_NAME_BASED_SHA1 = 5;
+
+BEGIN_EXTERN_C()
+
+PHPAPI void php_uuid_create_v3(php_uuid *uuid, const php_uuid *namespace, const char *name, const size_t name_len);
+
+PHPAPI int php_uuid_create_v4(php_uuid *uuid, const zend_bool throw);
+
+static zend_always_inline int php_uuid_create_v4_silent(php_uuid *uuid)
+{
+	return php_uuid_create_v4(uuid, 0);
+}
+
+static zend_always_inline int php_uuid_create_v4_throw(php_uuid *uuid)
+{
+	return php_uuid_create_v4(uuid, 1);
+}
+
+PHPAPI void php_uuid_create_v5(php_uuid *uuid, const php_uuid *namespace, const char *name, const size_t name_len);
+
+static const zend_always_inline php_uuid_variant php_uuid_get_variant(const php_uuid *uuid)
+{
+	if ((uuid->bytes[8] & 0xC0) == 0x80) return PHP_UUID_VARIANT_RFC4122;
+	if ((uuid->bytes[8] & 0xE0) == 0xC0) return PHP_UUID_VARIANT_MICROSOFT;
+	if ((uuid->bytes[8] & 0x80) == 0x00) return PHP_UUID_VARIANT_NCS;
+	return PHP_UUID_VARIANT_FUTURE_RESERVED;
+}
+
+static const zend_always_inline uint8_t php_uuid_get_version(const php_uuid *uuid)
+{
+	return uuid->bytes[6] >> 4;
+}
+
+static const zend_always_inline php_uuid php_uuid_namespace_dns()
+{
+	return (php_uuid) { "\x6b\xa7\xb8\x10\x9d\xad\x11\xd1\x80\xb4\x00\xc0\x4f\xd4\x30\xc8" };
+}
+
+static const zend_always_inline php_uuid php_uuid_namespace_oid()
+{
+	return (php_uuid) { "\x6b\xa7\xb8\x12\x9d\xad\x11\xd1\x80\xb4\x00\xc0\x4f\xd4\x30\xc8" };
+}
+
+static const zend_always_inline php_uuid php_uuid_namespace_url()
+{
+	return (php_uuid) { "\x6b\xa7\xb8\x11\x9d\xad\x11\xd1\x80\xb4\x00\xc0\x4f\xd4\x30\xc8" };
+}
+
+static const zend_always_inline php_uuid php_uuid_namespace_x500()
+{
+	return (php_uuid) { "\x6b\xa7\xb8\x14\x9d\xad\x11\xd1\x80\xb4\x00\xc0\x4f\xd4\x30\xc8" };
+}
+
+static const zend_always_inline php_uuid php_uuid_nil()
+{
+	return (php_uuid) { "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" };
+}
+
+static const zend_always_inline int php_uuid_is_nil(const php_uuid *uuid)
+{
+	const php_uuid nil = php_uuid_nil();
+
+	return memcmp(uuid->bytes, &nil, PHP_UUID_LEN) == 0;
+}
+
+PHPAPI int php_uuid_parse(php_uuid *uuid, const char *input, const size_t input_len, const zend_bool throw);
+
+static zend_always_inline int php_uuid_parse_silent(php_uuid *uuid, const char *input, const size_t input_len)
+{
+	return php_uuid_parse(uuid, input, input_len, 0);
+}
+
+static zend_always_inline int php_uuid_parse_throw(php_uuid *uuid, const char *input, const size_t input_len)
+{
+	return php_uuid_parse(uuid, input, input_len, 1);
+}
+
+static zend_always_inline void php_uuid_to_hex(php_uuid_hex *buffer, const php_uuid *uuid)
```

Comment:
I'd move these functions into the C file. They're not going to benefit from inlining.